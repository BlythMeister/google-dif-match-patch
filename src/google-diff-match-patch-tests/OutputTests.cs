using ApprovalTests;
using ApprovalTests.Reporters;
using DiffMatchPatch;
using NUnit.Framework;

namespace DiffMatchPatchTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class OutputTests
    {
        private const string expected = "start\n" +
                                        "line with some content\n" +
                                        "line with some more content\n" +
                                        "line with some \twhitespace\t content\n" +
                                        "unchanged content\n" +
                                        "removed line\n" +
                                        "end";

        private const string actual = "start\n" +
                                      "new line\n" +
                                      "line with some modified content\n" +
                                      "line with some more content\tthat has a tab added\n" +
                                      "line with some whitespace content\n" +
                                      "unchanged content\n" +
                                      "end";

        private const string expectedHtml = "<p>start</p>\n" +
                                            "<p>line with some content</p>\n" +
                                            "<p>line with some more content</p>\n" +
                                            "<p>line with some \twhitespace\t content</p>\n" +
                                            "<p>unchanged content</p>\n" +
                                            "<p>removed line</p>\n" +
                                            "<p>end</p>";

        private const string actualHtml = "<p>start</p>\n" +
                                          "<p>new line</p>\n" +
                                          "<p>line with some modified content</p>\n" +
                                          "<p>line with some more content\tthat has a tab added</p>\n" +
                                          "<p>line with some whitespace content</p>\n" +
                                          "<p>unchanged content</p>\n" +
                                          "<p>end</p>";

        [Test]
        public void CorrectPatchRawTextOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToText()));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchTextOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToReadableText()));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchHtmlOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToHtml(), "html"));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchHtmlDocOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToHtmlDocument(), "html"));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectDiffRawTextOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToText()));
        }

        [Test]
        public void CorrectDiffTextOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToReadableText()));
        }

        [Test]
        public void CorrectDiffHtmlOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToHtml(), "html"));
        }

        [Test]
        public void CorrectDiffHtmlDocOutput_TextInput()
        {
            var diffs = Diff.Compute(expected, actual);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToHtmlDocument(), "html"));
        }

        [Test]
        public void CorrectPatchRawTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToText()));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToReadableText()));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchHtmlEncodedTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToReadableText(true)));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchHtmlOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToHtml(), "html"));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectPatchHtmlDocOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            var patches = Patch.FromDiffs(diffs);

            Approvals.Verify(new ApprovalTextWriter(patches.ToHtmlDocument(), "html"));
            Assert.AreEqual(6, patches.Count);
        }

        [Test]
        public void CorrectDiffRawTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToText()));
        }

        [Test]
        public void CorrectDiffTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToReadableText()));
        }

        [Test]
        public void CorrectDiffHtmlEncodedTextOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToReadableText(true)));
        }

        [Test]
        public void CorrectDiffHtmlOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToHtml(), "html"));
        }

        [Test]
        public void CorrectDiffHtmlDocOutput_HtmlInput()
        {
            var diffs = Diff.Compute(expectedHtml, actualHtml);
            diffs.Cleanup();

            Approvals.Verify(new ApprovalTextWriter(diffs.ToHtmlDocument(), "html"));
        }
    }
}
