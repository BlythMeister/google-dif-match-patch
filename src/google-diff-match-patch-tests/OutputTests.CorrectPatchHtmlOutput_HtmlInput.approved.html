<!-- START OF DIFF STYLE -->
<style type="text/css">
.diffContainer {
	font-family: 'Courier New', Courier, monospace;
}

.diffInsert {
	background:#e6ffe6;
}

.diffDelete {
	background:#ffe6e6;
}
</style>
<!-- END OF DIFF STYLE -->
<!-- START OF PATCH STYLE -->
<style type="text/css">
.patchContainer {
	font-family: 'Courier New', Courier, monospace;
	border: 1px solid black;
	margin-bottom: 5px;
	padding: 5px;
}

.patchNote {
	font-weight: bold;
}

.patchTitle {
	padding: 5px 0;
}

.patchDiffs {
	border-top: 1px solid black; 
	padding: 5px 0;
}
</style>
<!-- END OF PATCH STYLE -->
<!-- START OF PATCHES -->
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 1</div>
<div>Delete character coordinates: 2,24</div>
<div>Insert character coordinates: 2,40</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">p&gt;start&lt;/p&gt;&#182;<br /></span><ins class="diffInsert">&lt;p&gt;new&#183;line&lt;/p&gt;&#182;<br /></ins><span class="diffUnchanged">&lt;p&gt;line&#183;with</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 2</div>
<div>Delete character coordinates: 40,16</div>
<div>Insert character coordinates: 40,25</div>
<div class="patchNote">NOTE: Patch coordinates are based on patch 1 being applied</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">th&#183;some&#183;</span><ins class="diffInsert">modified&#183;</ins><span class="diffUnchanged">content&lt;</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 3</div>
<div>Delete character coordinates: 83,32</div>
<div>Insert character coordinates: 83,53</div>
<div class="patchNote">NOTE: Patch coordinates are based on patch 2 being applied</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">ome&#183;more&#183;content</span><ins class="diffInsert">&#187;that&#183;has&#183;a&#183;tab&#183;added</ins><span class="diffUnchanged">&lt;/p&gt;&#182;<br />
&lt;p&gt;line&#183;wit</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 4</div>
<div>Delete character coordinates: 135,17</div>
<div>Insert character coordinates: 135,16</div>
<div class="patchNote">NOTE: Patch coordinates are based on patch 3 being applied</div>
<div class="patchNote">NOTE: Patch contains whitespace only differences</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">th&#183;some&#183;</span><del class="diffDelete">&#187;</del><span class="diffUnchanged">whitespa</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 5</div>
<div>Delete character coordinates: 145,17</div>
<div>Insert character coordinates: 145,16</div>
<div class="patchNote">NOTE: Patch coordinates are based on patch 4 being applied</div>
<div class="patchNote">NOTE: Patch contains whitespace only differences</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">itespace</span><del class="diffDelete">&#187;</del><span class="diffUnchanged">&#183;content</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<div class="patchContainer">
<div class="patchTitle">
<div>Patch number: 6</div>
<div>Delete character coordinates: 187,28</div>
<div>Insert character coordinates: 187,8</div>
<div class="patchNote">NOTE: Patch coordinates are based on patch 5 being applied</div>
</div>
<div class="patchDiffs">
<!-- START OF DIFFS -->
<div class="diffContainer">
<span class="diffUnchanged">/p&gt;&#182;<br /></span><del class="diffDelete">&lt;p&gt;removed&#183;line&lt;/p&gt;&#182;<br /></del><span class="diffUnchanged">&lt;p&gt;e</span>
</div>
<!-- END OF DIFFS -->
</div>
</div>
<!-- END OF PATCHES -->